test_runner/fixtures/neon_fixtures.py

```diff
@@ -4121,6 +4135,11 @@ def stop(self, mode: str = "fast") -> "Endpoint":
                 self.endpoint_id, check_return_code=self.check_stop_result, mode=mode
             )
 
+        if sks_wait_walreceiever_gone is not None:
```

Comment:
Should this be done even if the lock acquire failed (running)? For me it sounds like a deadlock hazard. Well, probably this is more related to the introduction of running lock..

Reply:
:shrug: Seems like the idea is that if lock is busy, then someone is already shutting down the compute and so it will complete soon.